libs/utils/analysis/latency_analysis.py

```diff
@@ -286,6 +286,42 @@ def cr(row):
         run_df = run_df[run_df.next_state.isin(['S', 'x'])][['running_time']]
         return run_df
 
+    @memoized
+    def _dfg_task_residency(self, task):
+        """
+        DataFrame of a task's execution time on each CPU
+
+        The returned DataFrame index is the CPU indexes
+        The DataFrame has just one column:
+        - runtime: the time the task spent being active (running or waking up)
+          on a given CPU, in seconds.
+
+        :param task: the task to report runtimes for
+        :type task: int or str
+        """
+        cpus = range(self._platform['cpus_count'])
+        runtimes = {cpu : 0.0 for cpu in cpus}
+
+        df = self._dfg_latency_df(task)
+
+        # Exclude sleep time
+        df = df[df.curr_state != 'S']
+
+        for time, data in df.iterrows():
+            cpu = data['__cpu']
+            if np.isnan(cpu):
```

Comment:
Can this really happen?!?
Since `__cpu` is a common field reporting the CPU in which an event has been generated... I would never expect this to be NaN.

Reply:
For wakeups, `__cpu` will be unset and `target_cpu` will be set instead:

target_cpu | __cpu | curr_state | next_state | t_start | t_delta
NaN | 6.0 | S | W | 2783.651695 | 0.010125
4.0 | NaN | W | A | 2783.661820 | 0.010525

Reply:
Oh right, indeed a wakeup can happen on CPU which is different then the one we wanna run the task on... I forgot there was this asymmetry in the latency dataframe.
Maybe we should also rename that "__cpu" in something like "wakeup_cpu"... just to avoid confusion in the future.

But for the time being, being add a line comment to remember that "__cpu" has a value only for "wakeup" events.